enhancements/installer/telemeter-metrics.md

```diff
@@ -0,0 +1,300 @@
+
+```
+// AddToMetric adds a sample to a metrics vector.
+AddToMetric(metric string, labels map[string]string, value string)
+```
+
+### Risks and Mitigations
+
+The pushgateway would be an a public endpoint reachable on the internet. Therefore some authentication or security should be in place in order to prevent malicious pushing of metrics.
+
+Users and customers may be wary of exposing information when using the installer.
+
+
+## Design Details
+
+### Test Plan
+
+**TODO**
+
+**Note:** *Section not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy. Anything
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).
+
+### Graduation Criteria
+
+**TODO**
+
+**Note:** *Section not required until targeted at a release.*
+
+Define graduation milestones.
+
+These may be defined in terms of API maturity, or as something else. Initial proposal
+should keep this high-level with a focus on what signals will be looked at to
+determine graduation.
+
+Consider the following in developing the graduation criteria for this
+enhancement:
+- Maturity levels - `Dev Preview`, `Tech Preview`, `GA`
+- Deprecation
+
+Clearly define what graduation means.
+
+#### Examples
+
+These are generalized examples to consider, in addition to the aforementioned
+[maturity levels][maturity-levels].
+
+##### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+
+##### Tech Preview -> GA 
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+##### Removing a deprecated feature
+
+- Announce deprecation and support policy of the existing feature
+- Deprecate the feature
+
+### Upgrade / Downgrade Strategy
+
+Not applicable.
+
+### Version Skew Strategy
+
+Not applicable. 
+
+## Implementation History
+
+Major milestones in the life cycle of a proposal should be tracked in `Implementation
+History`.
+
+## Drawbacks
+
+By using the aggregation pushgateway, we are losing specificity. We will not be able to tell whether one user heavily skews results.
+
+ We want to understand user events in how they interact with the installer and get specific error messages and logs, but we cannot do that with Prometheus as a backing service. 
```

Comment:
It's still a pretty long way to go I think, but there are plans to support log ingestion in the Observatorium system (the system that implements the metrics system today)